docs/howto/custom-template-backend.txt

```diff
@@ -0,0 +1,173 @@
+=======================
+Custom template backend
+=======================
```

Comment:
I would add this page also to the main index, e.g.

```diff
diff --git a/docs/index.txt b/docs/index.txt
index e55d28b22f..5f32c244cd 100644
--- a/docs/index.txt
+++ b/docs/index.txt
@@ -174,6 +174,7 @@ designers and how it can be extended by programmers:
 * **For programmers:**
   :doc:`Template API <ref/templates/api>` |
   :doc:`Custom tags and filters <howto/custom-template-tags>`
+  :doc:`Custom template backend <howto/custom-template-backend>`
 
 Forms
 =====
```